go/vt/schemadiff/errors.go

```diff
@@ -228,17 +228,70 @@ func (e *InvalidColumnInCheckConstraintError) Error() string {
 		sqlescape.EscapeID(e.Column), sqlescape.EscapeID(e.Constraint), sqlescape.EscapeID(e.Table))
 }
 
+type ForeignKeyDependencyUnresolvedError struct {
+	Table string
+}
+
+func (e *ForeignKeyDependencyUnresolvedError) Error() string {
+	return fmt.Sprintf("table %s has unresolved/loop foreign key dependencies",
+		sqlescape.EscapeID(e.Table))
+}
+
 type InvalidColumnInForeignKeyConstraintError struct {
 	Table      string
 	Constraint string
 	Column     string
 }
 
 func (e *InvalidColumnInForeignKeyConstraintError) Error() string {
-	return fmt.Sprintf("invalid column %s referenced by foreign key constraint %s in table %s",
+	return fmt.Sprintf("invalid column %s covered by foreign key constraint %s in table %s",
 		sqlescape.EscapeID(e.Column), sqlescape.EscapeID(e.Constraint), sqlescape.EscapeID(e.Table))
 }
 
+type InvalidReferencedColumnInForeignKeyConstraintError struct {
+	Table            string
+	Constraint       string
+	ReferencedTable  string
+	ReferencedColumn string
+}
+
+func (e *InvalidReferencedColumnInForeignKeyConstraintError) Error() string {
+	return fmt.Sprintf("invalid column %s.%s referenced by foreign key constraint %s in table %s",
+		sqlescape.EscapeID(e.ReferencedTable), sqlescape.EscapeID(e.ReferencedColumn), sqlescape.EscapeID(e.Constraint), sqlescape.EscapeID(e.Table))
+}
+
+type MismatchingForeignKeyColumnCountError struct {
```

Comment:
Name suggestion: `ForeignKeyColumnCountMismatchError`. I don't know if users ever see this though so it may not matter too much.

Reply:
Renamed